TheSecondJava

```diff
@@ -0,0 +1,39 @@
+TextView tvSonuc;
+TextView tvAciklama;
+private Button btn1;
+private Button btn2;
+
+float idealWeight = 0;
+float myAverage = 0;
+tvSonuc = (TextView) findViewById(R.id.textView1);
+tvAciklama = (TextView) findViewById(R.id.textView2);
+idealWeight = Float.parseFloat(getIntent().getStringExtra("myIdealWeight"));
+myAverage = Float.parseFloat(getIntent().getStringExtra("myCurrentAverage"));
+
+if(idealWeight > 0){
+    tvSonuc.setText(getString(R.string.idealWeight)+"; "+Math.round(idealWeight));
+}else{
+    tvSonuc.setText(getString(R.string.val001));
+}
+
+if(Math.round(myAverage) >0 && Math.round(myAverage)<18.49){
+    tvAciklama.setText(getString(R.string.val002));
+}else if(Math.round(myAverage) >18.49 && Math.round(myAverage)<24.99){
+    tvAciklama.setText(getString(R.string.val003));
+}else if(Math.round(myAverage) >24.99 && Math.round(myAverage)<29.99){
+    tvAciklama.setText(getString(R.string.val004));
+}else if(Math.round(myAverage) >29.99 && Math.round(myAverage)<34.99){
+    tvAciklama.setText(getString(R.string.val005));
+}else if(Math.round(myAverage) >34.99 && Math.round(myAverage)<44.99){
+    tvAciklama.setText(getString(R.string.val006));
+}else if(Math.round(myAverage) >44.99){
+    tvAciklama.setText(getString(R.string.val007));
+}
```

Comment:
It's better to have collection and iterate it instead of multiple if-s or use separate function with switch/case

Reply:
I think it is very difficult to get what you want using the Switch / Case method. As far as I remember, it is a bit difficult to define a variable in a code in this structure. Now I'm trying to fix the code.

Reply:
(It's better to have collection and iterate it instead of multiple if-s or use separate function with switch/case)
it's impossible to use a switch when you have checks like < or >, switch is only supporting exact values.I couldn't make this change.